packages/pipeline-editor/src/PipelineController/index.ts

```diff
@@ -444,10 +445,44 @@ class PipelineController extends CanvasController {
         nodes.push(...c.node_types);
       }
     }
+    nodes = this.propagatePipelineDefaultProperties(nodes, this.palette);
 
     return nodes;
   }
 
+  propagatePipelineDefaultProperties(
+    nodes: NodeType[],
+    palette: PaletteV3
+  ): NodeType[] {
+    return produce(nodes, (draft: any) => {
+      for (const prop of palette.properties?.uihints?.parameter_info ?? []) {
+        const propValue = this.getPipelineFlow()?.pipelines?.[0]?.app_data
+          ?.properties?.pipeline_defaults?.[
+          prop.parameter_ref.replace(/^elyra_/, "")
+        ];
+
+        draft.forEach((node: any) => {
+          const propIndex = node.app_data.properties.uihints.parameter_info.findIndex(
+            (p: any) => p.parameter_ref === prop.parameter_ref
+          );
+          const nodeProp =
+            node.app_data.properties.uihints.parameter_info[propIndex];
+          if (prop.custom_control_id === "EnumControl") {
+            const propLabel = nodeProp?.data?.labels?.[propValue] ?? propValue;
+            if (nodeProp && propLabel) {
+              nodeProp.data.placeholder = `${propLabel} (pipeline default)`;
+              nodeProp.data.pipeline_default = true;
+            }
+          } else if (prop.custom_control_id === "StringArrayControl") {
+            if (nodeProp) {
+              nodeProp.data.pipeline_defaults = propValue;
+            }
```

Comment:
Not breaking, just thought about rearranging this part a bit.
This nested if could be simplified to something like 
```suggestion
          } else if (prop.custom_control_id === "StringArrayControl" && nodeProp) {
            nodeProp.data.pipeline_defaults = propValue;

```
or have `if (nodeProp)` as a top check before looking into `custom_control_id`s?

Reply:
nice catch, I didn't notice how this ended up after all my iteration. I'll streamline the if checks in this function